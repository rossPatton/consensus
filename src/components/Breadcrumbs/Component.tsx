import React, { memo } from 'react';
import { Link } from 'react-router-dom';

export const BreadcrumbsComponent = memo((props: any) => (
  <ul className="fs6 fw600 mB2 lh1 lsNone fx fxWrap brdB1 mB3 pB3">
    {console.log('props => ', props)}
    {props.country && (
      <li className="mR1">
        {!props.region && props.country.name}
        {props.region && (
          <Link to={`/${props.country.code}`}>
            {props.country.name}
          </Link>
        )}
      </li>
    )}
    {props.region && (
      <li className="mHide mR1">/</li>
    )}
    {props.region && (
      <li className="mR1">
        {!props.city && props.country.name}
        {props.city && (
          <Link to={`/${props.country.code}/${props.region.code}`}>
            {props.region.name}
          </Link>
        )}
      </li>
    )}
    <li className="mHide mR1">/</li>
    <li className="mR1">
      <Link to="/us/ny/new-york-city">
        New York City
      </Link>
    </li>
  </ul>
));
