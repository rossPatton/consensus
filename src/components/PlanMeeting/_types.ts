import {Location} from 'history';

export type tState = Partial<tEvent> & {
  duration: string | number,
  time: string,
};

export type tStore = {
  eventsByOrgId: tThunk<tEvent[]>,
  session: tThunk<tSession>,
};

export type tStateUnion = keyof tState;
export type tEventTypes = React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>;

export type tContainerProps = {
  events: tEvent[],
  org: tOrg,
  postEvent: (query: tUpsertEventQuery) => tThunkPayload<tEvent>,
  router: Location,
  session: tSession,
};

export type tComponentProps = tContainerProps & tState & {
  onSubmit: (ev: React.MouseEvent<HTMLButtonElement>) => void,
  saveAsDraft: (ev: React.MouseEvent<HTMLButtonElement>) => void,
  updateState: (stateKey: tStateUnion, value: any) => void,
};
