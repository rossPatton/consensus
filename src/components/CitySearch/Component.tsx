import _ from 'lodash';
import React, {memo} from 'react';

import stateCodeMap from '../../json/usa/stateCodeMap.json';
import {tComponentProps} from './_types';

export const CitySearchComponent = memo((props: tComponentProps) => {
  const onCityChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const city = _.find(props.cities, c => c.name === ev.currentTarget.value);
    if (city) {
      props.updateState({
        city: city.name,
        cityId: city.id,
        region: city.region,
        regionId: city.regionId,
      });
    }
  };

  const onRegionChange = (ev: React.ChangeEvent<HTMLSelectElement>) => {
    const region = ev.currentTarget.value;
    if (region !== props.region) {
      props.updateState({
        city: '',
        cityId: 0,
        region,
        regionId: 0,
      });
    }
  };

  return (
    <>
      {props.showRegionField && (
        <>
          <h2 className="fs5 mB1 lh1">
            Pick a different state
          </h2>
          <label htmlFor="stateSelect">
            <select
              name="type"
              id="stateSelect"
              className="mB4 row bgWhite"
              value={props.region}
              onBlur={onRegionChange}
              onChange={onRegionChange}>
              {Object.keys(stateCodeMap).map(state => (
                <option key={state} value={state}>
                  {state}
                </option>
              ))}
            </select>
          </label>
        </>
      )}
      <h2 className="fs5 mB1 lh1">
        City in <span className="dInBl mR2">
          {(!props.showRegionField && props.geo.region) && props.geo.region}
          {(props.showRegionField && props.region) && props.region}
        </span>
        {!props.showRegionField && (
          <button
            type="button"
            onClick={props.toggleRegionField}>
            Not the right state?
          </button>
        )}
      </h2>
      {props.label && (
        <p className="fs5 copyBlack mB1">
          {props.label}
        </p>
      )}
      <div className="mB4">
        {props.cityId === 0 && (
          <label htmlFor="citySelect">
            <select
              name="type"
              id="citySelect"
              className="row bgWhite"
              value={props.city}
              onBlur={onCityChange}
              onChange={onCityChange}>
              {props.cities.map((city: tCity) => (
                <option key={city.id} value={city.name}>
                  {city.name}, {city.region}
                </option>
              ))}
            </select>
          </label>
        )}
        {props.cityId > 0 && (
          <div className="brdA1 p3 black br8 dInBl">
            <b>{props.city}</b>, <span className="dInBl mR3">{props.region}</span>
            <button
              type="button"
              onClick={() => {
                props.updateState({
                  city: '',
                  cityId: 0,
                });
              }}>
              Not the right city?
            </button>
          </div>
        )}
      </div>
    </>
  );
});
