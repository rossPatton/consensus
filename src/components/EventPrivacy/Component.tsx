import React, {memo} from 'react';

import {tProps} from './_types';

export const EventPrivacyComponent = memo((props: tProps) => (
  <small className="bgYellowLite br8 p1 pL2 pR2 mR2">
    {props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Lock Emoji">
          ðŸ”’
        </span>
        Private Event
      </>
    )}
    {!props.isPrivate && (
      <>
        <span
          role="img"
          className="mR1"
          aria-label="Tada Emoji">
          ðŸŽ‰
        </span>
        Public Event
      </>
    )}
  </small>
));
