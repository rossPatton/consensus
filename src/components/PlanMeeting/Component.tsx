import {objToQueryString} from '@app/utils';
import dayJS from 'dayjs';
import _ from 'lodash';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {tComponentProps} from './_types';

export const PlanMeetingComponent = memo((props: tComponentProps) => {
  const {updateState} = props;
  const onDurationChange = (ev: React.ChangeEvent<HTMLSelectElement>) =>
    updateState(
      'duration',
      parseInt(ev.currentTarget.value, 10),
    );

  const disableSubmit = !props.title
    || !props.date
    || !props.duration
    || !props.time;

  return (
    <div className="bg-white rounded p-2">
      {props.heading && (
        <h1 className="text-3 mb-2">
          {props.isDraft && 'Edit Meeting Draft'}
          {props.isCopy && 'Copy Meeting'}
          {!props.isCopy && !props.isDraft && 'Plan Meeting'}
        </h1>
      )}
      <form
        id="form"
        method="POST"
        encType="multipart/form-data">
        <fieldset>
          <h3 className="mb-1">Meeting Title</h3>
          <input
            className="mb-3 w-full"
            placeholder="Your Meeting Title Here"
            value={props.title}
            onChange={ev => updateState('title', ev.currentTarget.value)}
          />
          <h3 className="mb-1">Tell Us About Your Meeting</h3>
          <textarea
            rows={8}
            spellCheck
            className="mb-3 w-full"
            placeholder="Meeting Description Here"
            value={props.description}
            onChange={ev => updateState('description', ev.currentTarget.value)}
          />
          <h3 className="mb-1">Where is your meeting happening?</h3>
          <div className="flex flex-col d:flex-row mb-3">
            <input
              results={3}
              spellCheck
              className="w-full mb-1 d:mb-0 mr-2"
              placeholder="The name or address of the place"
              value={props.location}
              onChange={ev => updateState('location', ev.currentTarget.value)}
            />
            <input
              type="url"
              className="w-full"
              placeholder="https://example.com"
              value={props.locationLink}
              onChange={ev => updateState('locationLink', ev.currentTarget.value)}
            />
          </div>
          <div className="flex flex-col d:flex-row mb-3">
            <div className="mb-3 d:mb-0 mr-2">
              <h3 className="mb-1">When&apos;s the meeting?</h3>
              <input
                type="date"
                min={dayJS().toISOString()}
                className="mr-1"
                value={dayJS(props.date).format('YYYY-MM-DD')}
                placeholder="Meeting Date Here"
                onChange={ev => updateState('date', ev.currentTarget.value)}
              />
              <input
                type="time"
                value={props.time}
                placeholder="Meeting Time Here"
                onChange={ev => updateState('time', ev.currentTarget.value)}
              />
            </div>
            <div>
              <h3 className="mb-1">How long is it?</h3>
              <select
                value={props.duration}
                onBlur={onDurationChange}
                onChange={onDurationChange}>
                <option value="1">1 hour</option>
                <option value="2">2 hours</option>
                <option value="3">3 hours</option>
                <option value="custom">Set an end time</option>
              </select>
            </div>
          </div>
          <div className="flex items-center">
            <button
              className="p-2 mr-1 hover:bg-gray-3"
              disabled={disableSubmit}
              onClick={ev => {
                ev.preventDefault();
                props.onSubmit();
              }}>
              Publish
            </button>
            <button
              className="p-2 mr-1 hover:bg-gray-3"
              disabled={disableSubmit}
              onClick={ev => {
                ev.preventDefault();
                props.saveAsDraft();
              }}>
              Save as Draft
            </button>
            {!isNaN(props.id) && (
              <Link
                to={`/draft/${props.id}?${objToQueryString(props)}`}
                className="btn p-2 text-sm hover:bg-gray-3">
                <span
                  role="img"
                  aria-label="Eye Emoji">
                  üëÅÔ∏è
                </span>
                Preview
              </Link>
            )}
          </div>
        </fieldset>
      </form>
    </div>
  );
});
