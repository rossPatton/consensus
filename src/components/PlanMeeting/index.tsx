import dayJS from 'dayjs';
import _ from 'lodash';
import loglevel from 'loglevel';
import qs from 'query-string';
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router';

import {Helmet} from '~app/components';
import {ErrorBoundary} from '~app/containers';
import {getMeetingsByGroupIdSuccess, patchEvent, postMeeting} from '~app/redux';
import {parseTimeString} from '~app/utils';

import {tContainerProps, tKeyUnion, tState, tStore, tValueUnion} from './_types';
import {PlanMeetingComponent} from './Component';

// @TODO this is a sub-page of a couple routes, should this be in /components???
class PlanMeetingContainer extends Component<tContainerProps, tState> {
  state = {
    category: this.props.group.category,
    cityId: this.props.group.cityId,
    date: dayJS().toISOString(),
    description: '',
    duration: 2,
    id: null as number | null,
    isCopy: false,
    isDraft: false,
    isPrivate: this.props.group.type !== 'public',
    location: '',
    locationLink: '',
    groupName: this.props.group.name,
    time: '19:00',
    title: '',
  };

  // we use query params to populate the form when editing an meeting
  constructor(props: tContainerProps) {
    super(props);
    const {router: {search}} = props;
    const draft = qs.parse(search);

    if (_.isEmpty(draft)) return;

    const isPrivate = draft.isPrivate === 'true';
    const isCopy = draft.isCopy === 'true';

    // if draft is a copy (to make new meeting), not an edit (of existing draft or meeting)
    // then we change a few things here
    const state = {
      category: draft.category as ts.category,
      cityId: typeof draft.cityId === 'string'
        ? parseInt(draft.cityId as string, 10)
        : null,
      // convert UTC date with tz to local format for html5 date/time picker
      date: isCopy
        ? dayJS().toISOString()
        : dayJS(draft.date as string).format('YYYY-MM-DD'),
      description: draft.description as string,
      duration: isCopy ? 2 : parseInt(draft.duration as string, 10),
      id: isCopy ? null : parseInt(draft.id as string, 10),
      isCopy: isCopy,
      isDraft: true,
      isPrivate,
      location: draft.location as string,
      locationLink: draft.locationLink as string,
      groupName: this.props.group.name,
      time: isCopy ? '19:00' : draft.time as string,
      title: draft.title as string,
    };

    this.state = state;
  }

  saveAsDraft = () =>
    this.setState({
      isDraft: true,
    }, () => this.onSubmit(true));

  onSubmit = async (saveAsDraft: boolean = false) => {
    const {duration, time, ...restOfEvent} = this.state;

    const timeArr = parseTimeString(time);
    const dur = typeof duration === 'string' ? parseInt(duration, 10) : duration;
    const date = dayJS(this.state.date).hour(timeArr?.[0]).minute(timeArr[1]);
    const endDate = dayJS(this.state.date).hour(timeArr?.[0]).minute(timeArr[1]);
    endDate.hour(endDate.hour() + dur);

    let newEvent: ts.meeting;
    try {
      const {patchEventDispatch, postMeetingDispatch} = this.props;

      // if meeting has already been saved as a draft, we will have the id
      // patch in that case, else post new meeting (initial draft save, or submit)
      const dispatch = this.state.id
        ? patchEventDispatch
        : postMeetingDispatch;

      const planMeeting = await dispatch({
        ...restOfEvent,
        // we submit drafts to the same table in the DB as well
        // we only want to save as draft when the user hits the save as draft button
        // we don't use state here, since we want to set this to false when publishing
        isDraft: saveAsDraft,
        // every date is stored in the db as an ISO string
        date: date.toISOString(),
        endDate: endDate.toISOString(),
        groupId: this.props.group.id as number,
      });

      newEvent = planMeeting.payload;
    } catch (err) {
      return loglevel.error('failed to save meeting to db', err);
    }

    // update redux on client side on meeting upsert success
    getMeetingsByGroupIdSuccess([newEvent, ...this.props.meetings]);

    // this will cause the preview button to render
    this.setState({
      id: newEvent.id,
    });
  }

  updateState = (stateKey: tKeyUnion, value: tValueUnion) => {
    this.setState({
      [stateKey]: value,
    } as Pick<tState, tKeyUnion>);
  }

  render() {
    const {group, sessionThunk} = this.props;

    return (
      <ErrorBoundary
        isSubPage
        status={sessionThunk?.error?.status}>
        <Helmet
          canonical={`/group/${group.handle}/planMeeting`}
          title=""
          meta={[
            { name: 'description', content: `Plan your next meeting with ${group.name}` },
            { name: 'keywords', content: 'plan,meeting' },
            { property: 'og:title', content: `Consensus: Plan Meeting for ${group.name}` },
            { property: 'og:description', content: `Plan your next meeting with ${group.name}` },
          ]}
        />
        {!sessionThunk.data.isAuthenticated && <Redirect to="" />}
        {sessionThunk.data.isAuthenticated && (
          <PlanMeetingComponent
            {...this.props}
            {...this.state}
            onSubmit={this.onSubmit}
            saveAsDraft={this.saveAsDraft}
            updateState={this.updateState}
          />
        )}
      </ErrorBoundary>
    );
  }
}

const mapStateToProps = (store: tStore) => ({
  meetings: store.meetingsByGroupId.data,
  sessionThunk: store.session,
});

const mapDispatchToProps = (dispatch: Function) => ({
  patchEventDispatch: (query: ts.upsertMeetingQuery) => dispatch(patchEvent(query)),
  postMeetingDispatch: (query: ts.upsertMeetingQuery) => dispatch(postMeeting(query)),
});

const PlanMeeting = connect(
  mapStateToProps,
  mapDispatchToProps,
)(PlanMeetingContainer);

export default PlanMeeting;
