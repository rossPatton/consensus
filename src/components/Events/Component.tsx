import cx from 'classnames';
import dayJS from 'dayjs';
import _ from 'lodash';
import qs from 'query-string';
import React, {memo} from 'react';
import {Link} from 'react-router-dom';

import {ExternalLink, PlaceholderImage, RSVP} from '../../components';
import {objToQueryString} from '../../utils';
import {tComponentProps} from './_types';

export const EventsComponent = memo((props: tComponentProps) => (
  <ul
    className={cx({
      'fx fxdRow taL': props.horizontal,
      'jcBetween': props.horizontal && props.events.length === 4,
    })}>
    {props.events.map((ev, i) => {
      const isPastEvent = dayJS(ev.date).isBefore(dayJS());

      return (
        <li
          key={ev.id}
          className={cx({
            'mB2': !props.horizontal,
            'col row': props.horizontal,
            'mR4': props.horizontal && i !== props.events.length - 1,
          })}>
          <div
            className={cx({
              'fx aiCtr': !props.horizontal,
              'p3 hvrBgGrey1 br8': props.isEditable,
            })}>
            <div
              className={cx({
                'mB2 bgWhite': props.horizontal,
                'mR3 bgGrey2': !props.horizontal,
              })}>
              <Link
                className="noUnderline"
                to={ev.isDraft
                  ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                  : `/event/${ev.id}`}>
                <PlaceholderImage
                  height={100}
                  seed={ev.id}
                  width={200}
                />
              </Link>
            </div>
            <div>
              <div
                className={cx({
                  'fx aiCtr mB1 fs7 fw600 lh1': true,
                  o5: isPastEvent,
                })}>
                <time className="mR1" dateTime={ev.date}>
                  {dayJS(ev.date).format('MMM DD | h:mmA')}
                </time>
                {!props.horizontal && (
                  <>
                    <span className="mR1">@</span>
                    {ev.locationLink && (
                      <ExternalLink
                        noFollow
                        className="mR1"
                        to={ev.locationLink}>
                        {ev.location}
                      </ExternalLink>
                    )}
                  </>
                )}
                {!ev.locationLink && ev.location}
              </div>
              <h3
                className={cx({
                  'fx aiCtr ttCap': true,
                  mB1: props.isEditable,
                  fs4: props.horizontal,
                })}>
                <Link
                  className="noUnderline"
                  to={ev.isDraft
                    ? `/org/${ev.orgId}/planMeeting?${qs.stringify(ev)}`
                    : `/event/${ev.id}`}>
                  {ev.title}
                </Link>
              </h3>
              {props.showRSVPs && (
                <div
                  className={cx({
                    'fx aiCtr fs6 lh1': true,
                    hide: props.horizontal,
                    o5: isPastEvent,
                  })}>
                  {!ev.isDraft && <RSVP event={ev} />}
                  {ev.isDraft && 'This event is not published yet. Click to edit.'}
                </div>
              )}
              {props.showOrgName
                && (
                  <Link
                    to={`/org/${ev.orgId}`}
                    className={cx({
                      'fw600 fs6 lh1': true,
                      o5: isPastEvent,
                    })}>
                    {ev.orgName}
                  </Link>
                )}
              {!isPastEvent
              && props.isEditable
              && (
                <div className="fx aiCtr mB1">
                  <div className="bgGrey4 br4 fs7 lh1 mR2 p1 pL2 pR2 white">
                    {ev.isDraft ? 'Draft' : 'Published'}
                  </div>
                  <div className="fx aiCtr mR1">
                    <Link
                      to={`/org/${ev.orgId}/planMeeting?${objToQueryString(ev)}`}
                      className="btn fs7 fw600 hvrBgGrey1 lh1 noUnderline p1 pL2 pR2">
                      <span
                        role="img"
                        aria-label="Hand with Pen Emoji">
                        ✍️
                      </span>
                      Edit
                    </Link>
                  </div>
                  <button
                    onClick={e => props.deleteEvent(e, ev.id)}
                    className="aiCtr fs7 fw600 fx hvrBgGrey1 lh1">
                    <span
                      role="img"
                      className="mR1"
                      aria-label="Big X Emoji">
                      ✖️
                    </span>
                    Delete
                  </button>
                </div>
              )}
            </div>
          </div>
        </li>
      );
    })}
  </ul>
));
