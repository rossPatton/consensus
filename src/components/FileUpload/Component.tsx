import cx from 'classnames';
import React, {memo} from 'react';
import {v1} from 'uuid';

import {Emoji} from '~app/components';

import {tComponentProps} from './_types';

const baseUrl = 'https://consensus.nyc3.digitaloceanspaces.com';

export const FileUploadComponent = memo((props: tComponentProps) => (
  <>
    <h3 className="text-base mb-1">
      {props.title}
    </h3>
    <div
      id="imagePreview"
      className={cx({
        'animated bg-gray-1 p-3 mb-3 flex flex-col items-center': true,
        'fadeInDown': props.img,
        'hidden': !props.img,
      })}>
      <button
        type="button"
        onClick={props.removeImage}
        className="mb-2 p-2">
        <Emoji
          label="Close Emoji"
          emoji="✖️"
        />
        Remove Image
      </button>
      {props.img && (
        <img
          alt="Your uploaded file preview."
          height={props.width}
          width={props.width}
          src={`${baseUrl}/${props.folder}/${props.img}${__DEV__ ? `?v=${v1()}` : ''}`}
          className={cx({
            'rounded w-full': !props.fieldKey.includes('Avatar'),
            'rounded-circ': props.fieldKey.includes('Avatar'),
          })}
        />
      )}
    </div>
    <label
      htmlFor="fileUpload"
      className={cx({
        'fadeInDown bg-gray-1 border cursor-ptr flex flex-col items-center mb-3 p-3 rounded animated': true,
        'fadeInDown': !props.img,
        'hidden': props.img,
      })}>
      <input
        accept="image/png, image/jpeg, image/gif"
        id="fileUpload"
        name="eventFeaturedImage"
        onChange={props.setImage}
        type="file"
        className={cx({
          hidden: true,
        })}
      />
      <span className="btn flex items-center mb-1 p-1">
        <Emoji
          label="Picture (Camera) Emoji"
          emoji="📷"
        />
        Upload Your Image
      </span>
      {props.info && (
        <small>{props.info}</small>
      )}
    </label>
  </>
));
