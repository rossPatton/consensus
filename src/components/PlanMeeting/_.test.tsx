import React from 'react';
import { Provider } from 'react-redux';
import render from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { testMeeting1 } from '~app/constants/jest';

import PlanMeeting from '.';
import {PlanMeetingComponent} from './Component';

const mockStore = configureStore();
const testGroup = {
  category: 'Political',
  cityId: 1,
  name: 'Test Group',
  type: 'public',
} as ts.group;

const meetingThunk = {
  error: {message: 'test', status: 200 as ts.statusUnion},
  isLoading: false,
  data: testMeeting1,
};

describe('components/PlanMeeting', () => {
  it('container renders', () => {
    const store = mockStore({
      meetingsByGroupId: {data: [{}]},
      session: {data: {isAuthenticated: false}},
      uploads: {data: {meetingFeaturedImage: ''}},
    });

    const component = render.create((
      <Provider store={store}>
        <PlanMeeting
          group={testGroup}
          router={{pathname: '', state: {}, hash: '', search: ''}}
        />
      </Provider>
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders component without crashing', () => {
    const component = render.create((
      <PlanMeetingComponent
        group={testGroup}
        img=""
        endTime="9:00"
        error="error"
        isCopy={false}
        isOnline
        time="7:00"
        meetingThunk={meetingThunk}
        onSubmit={jest.fn()}
        saveAsDraft={jest.fn()}
        updateState={jest.fn()}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders component as draft', () => {
    const component = render.create((
      <PlanMeetingComponent
        group={testGroup}
        img=""
        endTime="9:00"
        error="error"
        isCopy={false}
        isDraft
        isOnline={false}
        time="7:00"
        meetingThunk={meetingThunk}
        onSubmit={jest.fn()}
        saveAsDraft={jest.fn()}
        updateState={jest.fn()}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders component as draft and copy', () => {
    const component = render.create((
      <PlanMeetingComponent
        group={testGroup}
        img=""
        endTime="9:00"
        error="error"
        isCopy
        isDraft
        isOnline
        time="7:00"
        meetingThunk={meetingThunk}
        onSubmit={jest.fn()}
        saveAsDraft={jest.fn()}
        updateState={jest.fn()}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders component loading state', () => {
    const component = render.create((
      <PlanMeetingComponent
        group={testGroup}
        img=""
        endTime="9:00"
        error="error"
        isCopy
        isDraft
        isOnline
        time="7:00"
        meetingThunk={{
          isLoading: true,
          ...meetingThunk,
        }}
        onSubmit={jest.fn()}
        saveAsDraft={jest.fn()}
        updateState={jest.fn()}
      />
    ));

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
