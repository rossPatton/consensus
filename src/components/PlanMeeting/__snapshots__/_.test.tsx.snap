// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/PlanMeeting container renders 1`] = `null`;

exports[`components/PlanMeeting renders component as draft 1`] = `
<div
  className="d:pb-2 d:pl-2 d:pr-2"
/>
`;

exports[`components/PlanMeeting renders component as draft and copy 1`] = `
<div
  className="d:pb-2 d:pl-2 d:pr-2"
/>
`;

exports[`components/PlanMeeting renders component loading state 1`] = `
<div
  className="d:pb-2 d:pl-2 d:pr-2"
/>
`;

exports[`components/PlanMeeting renders component without crashing 1`] = `
<div
  className="d:pb-2 d:pl-2 d:pr-2"
/>
`;
